test/helper.py

```diff
@@ -75,19 +80,30 @@ def admin_user(self):
 
     def _get_admin(self):
         if not self._admin:
+            print("Logging on as Admin once")
             self._admin = UserAPI(self.config.api_server, self.config.admin_username, self.config.admin_password).login()
         return self._admin
 
     def _print_actual_name(self, name):
         print("  ({})".format(name))
 
-    def _check_admin_exists(self):
+    def _check_admin_and_group_exist(self):
         try:
             self._get_admin()
         except Exception:
             # create admin
+            print('Creating admin user')
             UserAPI(self.config.api_server, self.config.admin_username, self.config.admin_password).register_as_new()
-            self.admin_user().create_group(self.config.default_group)
-            
+
+        allGroups = self._get_admin().get_groups_as_string()
```

Comment:
It would be much cleaner if there was a get_groups() which returns a Python list containing the group names. It would then be trivial to sort them and extract the highest suffix used and start generating new names from there.